src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs

```diff
@@ -209,10 +209,26 @@ private static InputOperation UpdateOperation(InputOperation operation, string c
             }
             operationsMap.Add(operation, updatedOperation);
 
+            if (!updatedOperation.Parameters.Any(p => p.IsEndpoint || (p.Location == RequestLocation.Uri && p.Kind == InputOperationParameterKind.Client)))
```

Comment:
Suggest to group this into `UpdateOperationParameters`, in their include all the update logic for the parameter list.

Reply:
That is for paging operations. We don't call it.

Reply:
It is better to combine all those in the `UpdateOperationParameters` for all operations (paging or non-paging), and in this method, switch the operation type to apply needed changes, and keep common changes for all type operation.
Otherwise, there are duplicate code and spread changes in several places, hard to maintain

Reply:
Synced offline. We could improve this part later.